src/build123d/topology.py

```diff
@@ -3873,111 +4185,99 @@ def face(self) -> Face:
         return solids[0]
 
     def compounds(self) -> ShapeList[Compound]:
         """compounds - all the compounds in this ShapeList"""
-        return ShapeList([c for shape in self for c in shape.compounds()])
+        return ShapeList([c for shape in self for c in shape.compounds()])  # type: ignore
 
     def compound(self) -> Compound:
         """Return the Compound"""
         compounds = self.compounds()
         compound_count = len(compounds)
         if compound_count != 1:
             warnings.warn(
-                f"Found {compound_count} compounds, returning first",
-                stacklevel=2,
+                f"Found {compound_count} compounds, returning first", stacklevel=2
             )
         return compounds[0]
 
-    def __gt__(self, sort_by: Union[Axis, SortBy] = Axis.Z):
+    def __gt__(self, sort_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:  # type: ignore
         """Sort operator >"""
         return self.sort_by(sort_by)
 
-    def __lt__(self, sort_by: Union[Axis, SortBy] = Axis.Z):
+    def __lt__(self, sort_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:  # type: ignore
         """Reverse sort operator <"""
         return self.sort_by(sort_by, reverse=True)
 
-    def __rshift__(self, group_by: Union[Axis, SortBy] = Axis.Z):
+    def __rshift__(self, group_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:
         """Group and select largest group operator >>"""
         return self.group_by(group_by)[-1]
 
-    def __lshift__(self, group_by: Union[Axis, SortBy] = Axis.Z):
+    def __lshift__(self, group_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:
         """Group and select smallest group operator <<"""
         return self.group_by(group_by)[0]
 
-    def __or__(self, filter_by: Union[Axis, GeomType] = Axis.Z):
+    def __or__(self, filter_by: Union[Axis, GeomType] = Axis.Z) -> ShapeList[T]:
         """Filter by axis or geomtype operator |"""
         return self.filter_by(filter_by)
 
-    def __eq__(self, other: object):
+    def __eq__(self, other: object) -> bool:
         """ShapeLists equality operator =="""
         return (
-            set(self) == set(other) if isinstance(other, ShapeList) else NotImplemented
+            set(self) == set(other) if isinstance(other, ShapeList) else NotImplemented  # type: ignore
         )
 
     # Normally implementing __eq__ is enough, but ShapeList subclasses list,
     # which already implements __ne__, so we need to override it, too
-    def __ne__(self, other: ShapeList):
+    def __ne__(self, other: ShapeList) -> bool:  # type: ignore
         """ShapeLists inequality operator !="""
         return (
             set(self) != set(other) if isinstance(other, ShapeList) else NotImplemented
         )
 
-    def __add__(self, other: ShapeList):
+    def __add__(self, other: ShapeList) -> ShapeList[T]:  # type: ignore
         """Combine two ShapeLists together operator +"""
         return ShapeList(list(self) + list(other))
```

Comment:
unnecessary to convert to list + concat only to convert to `ShapeList`
```suggestion
        return ShapeList(chain(self, other))
```

Reply:
Done.

Reply:
... and undone.  MacOS-13 (Python 3.12 IIRC) fails with this change; however, it works everywhere else.  More investigation is required.